apps/api/scripts/calculate-next-season-eligibility.ts

```diff
@@ -235,37 +245,99 @@ Examples:
     );
 
-    // Step 6: Calculate individual rewards
+    // Step 6: Query for all agent boosters during the season.
+    const [season] = await db
+      .select()
+      .from(seasons)
+      .where(eq(seasons.number, seasonNumber));
+    if (!season) {
+      console.error(`Season ${seasonNumber} not found`);
+      process.exit(1);
+    }
+    const userAgentBoosts = await db
+      .select({
+        address: boostChanges.wallet,
+        boostAmount: sum(boostChanges.deltaAmount),
+      })
+      .from(agentBoosts)
+      .innerJoin(boostChanges, eq(agentBoosts.changeId, boostChanges.id))
+      .where(gte(agentBoosts.createdAt, season.startDate))
+      .groupBy(boostChanges.wallet);
+
+    const userAgentBoostsMap = userAgentBoosts.reduce(
+      (acc, boost) =>
+        acc.set(
+          BlockchainAddressAsU8A.decode(boost.address),
+          BigInt(boost.boostAmount || "0"),
+        ),
+      new Map<string, bigint>(),
+    );
+
+    // Step 7: Query for all users that had an agent compete in a competition during the season.
+    const userCompetitions = await db
+      .select({
+        address: users.walletAddress,
+        competitionIds: sql<
+          string[]
+        >`array_agg(DISTINCT ${competitions.id})`.as("competitionIds"),
+      })
+      .from(users)
+      .innerJoin(agents, eq(users.id, agents.ownerId))
+      .innerJoin(competitionAgents, eq(agents.id, competitionAgents.agentId))
+      .innerJoin(
+        competitions,
+        eq(competitionAgents.competitionId, competitions.id),
+      )
+      .where(
+        and(
+          gte(competitions.startDate, season.startDate),
+          lte(competitions.startDate, referenceTime),
+          eq(competitionAgents.status, "active"),
+        ),
+      )
+      .groupBy(users.walletAddress);
+
+    const userCompetitionsMap = userCompetitions.reduce(
+      (acc, userCompetition) =>
+        acc.set(userCompetition.address, userCompetition.competitionIds),
+      new Map<string, string[]>(),
+    );
+
+    // Step 7: Calculate individual rewards
     console.log(
       `${colors.blue}Step 6: Calculating individual rewards...${colors.reset}`,
     );
 
-    const eligibleAccounts: EligibleAccount[] = [];
+    const eligibilityEntries: EligibilityEntry[] = [];
 
     if (totalActiveStakes > 0n && availableRewards > 0n) {
       for (const [address, activeStake] of accountStakes.entries()) {
+        const isEligible =
+          userAgentBoostsMap.has(address) || userCompetitionsMap.has(address);
+
         // Calculate proportional reward: (account_stake / total_stakes) * available_rewards
         const reward = (activeStake * availableRewards) / totalActiveStakes;
 
-        eligibleAccounts.push({
+        eligibilityEntries.push({
           address,
           activeStake,
           reward,
+          ineligibleReason: isEligible
+            ? undefined
+            : "no boost or competition activity",
```

Comment:
Making this the flag so we know they are not eligible. We need to be sure that everything downstream from here works as expected (the genreate merkle tree script handling ineligible, whatever fuul does with the csv when we upload it, the web front end, etc).

I leave the reward amount just so we can show the user what they were eligible for when we display the ineligible allocation.